src/perl/nightly/NightlyRunRules.pm

```diff
@@ -135,57 +127,25 @@ sub _runPerlTests {
 }
 
 ######################################################################
-# Run the perl VDO tests. The targets are sets of tests defined in
-# vdotests.suites. The default target is the default (nightly) test suite.
-#
-# @param  properties  A hashref of runtime properties
-# @param  testName    A descriptive name for the tests being run
-# @param  osClass     The default OS class to reserve machines from
-# @oparam suite       The VDO test suite to run
-# @oparam otherArgs   Other arguments to pass to the tests
-##
-sub _runVDOTestSuite {
-  my ($properties, $testName, $osClass, $suite, $otherArgs)
-    = assertMinMaxArgs(['', ''], 3, 5, @_);
-
-  my $logDir = "$properties->{runLogDir}/$testName/";
-  my $rundir = "$properties->{perlRoot}/vdotest";
-  my $regexp = "The following tests failed";
-
-  my $cmd = "./vdotests.pl $suite --log=1 --JSON=1"
-            . " --logDir=$logDir"
-            . " --xmlOutput=1"
-            . " --saveServerLogDir=$logDir"
-            . " --nightlyStart=$properties->{nightlyStart}"
-            . " --moveToMaintenance --rsvpOSClass=$osClass $otherArgs";
-
-  if ($properties->{skipTestExecution}) {
-    $log->debug("Skipping execution of $testName with command: $cmd");
-    return;
-  }
-  NightlyUtils::runTests($testName, $cmd, $rundir, $logDir, $regexp);
-  installFiles("$logDir/*.xml", "$properties->{xmlLogDir}/vdotest");
-}
-
-######################################################################
-# Run the perl UDS tests. The targets are sets of tests defined in
-# udstests.suites. The default target is the default test suite.
+# Run the non-unit perl tests. If no suite is provided, use the default
+# (nightly) test suite.
 #
 # @param  properties  A hashref of runtime properties
 # @param  testName    A descriptive name for the tests being run
+# @param  type        The test type (vdo, uds, or dm)
 # @param  osClass     The default OS class to reserve machines from
-# @oparam suite       The UDS test suite to run
+# @oparam suite       The test suite to run
 # @oparam otherArgs   Other arguments to pass to the tests
 ##
-sub _runUDSTestSuite {
-  my ($properties, $testName, $osClass, $suite, $otherArgs)
-    = assertMinMaxArgs(['', ''], 3, 5, @_);
+sub _runTestSuite {
+  my ($properties, $testName, $type, $osClass, $suite, $otherArgs)
+    = assertMinMaxArgs(['', ''], 4, 6, @_);
 
   my $logDir = "$properties->{runLogDir}/$testName/";
-  my $rundir = "$properties->{perlRoot}/udstest";
+  my $rundir = "$properties->{perlRoot}/" . $type . "test";
```

Comment:
wouldn't /${type}test be simpler than mixing string interpolation and concatenation?

Reply:
I wasn't sure that interpolation would work correctly. It's plausibly simpler to read, though.

Reply:
This seems to work okay so I've updated the change to use interpolation instead.